search/includes/classes/commands/class-corecommand.php

```diff
@@ -14,38 +14,105 @@ class CoreCommand extends \ElasticPress\Command {
 	private const SUCCESS_ICON = "\u{2705}"; // unicode check mark
 	private const FAILURE_ICON = "\u{274C}"; // unicode cross mark
 
-	protected function _maybe_setup_index_version( &$assoc_args ) {
-		if ( array_key_exists( 'version', $assoc_args ) ) {
-			if ( $assoc_args['version'] ) {
-				$version_number = $assoc_args['version'];
+	private function _verify_arguments_compatibility( $assoc_args ) {
+		if ( array_key_exists( 'version', $assoc_args ) && array_key_exists( 'using-versions', $assoc_args ) ) {
+			return WP_CLI::error( 'The --version argument is not allowed when specifying --using-versions' );
+		}
 
-				// If version is specified, the indexable must also be specified, as different indexables can have different versions
-				if ( ! isset( $assoc_args['indexables'] ) ) {
-					return WP_CLI::error( 'The --indexables argument is required when specifying --version, as each indexable has separate versioning' );
-				}
+		// If version is specified, the indexable must also be specified, as different indexables can have different versions
+		if ( array_key_exists( 'version', $assoc_args ) && ! isset( $assoc_args['indexables'] ) ) {
+			return WP_CLI::error( 'The --indexables argument is required when specifying --version, as each indexable has separate versioning' );
+		}
+
+		if ( array_key_exists( 'using-versions', $assoc_args ) && ! isset( $assoc_args['indexables'] ) ) {
+			return WP_CLI::error( 'The --indexables argument is required when specifying --using-versions' );
+		}
+
+		if ( array_key_exists( 'using-versions', $assoc_args ) && isset( $assoc_args['network-wide'] ) ) {
+			return WP_CLI::error( 'The --using-versions argument is not supported together with --network-wide' );
+		}
+	}
```

Comment:
WP_CLI::error doesn't return (it either exits or throws an exception, so this is a bit misleading.

Reply:
There are other instances of this that should be addressed too.

Reply:
Good point. I removed it from this file.